### PR TITLE
#70 Full text search

### PR DESCRIPTION
**Pull request by lukybil**
- added generic `Autocomplete`
- added geocoding search to BottomSheet

### CODE REVIEW

File: components/inputs/Autocomplete.tsx [hunk 1]
```diff
@@ -39,39 +47,43 @@ function AutocompleteInner<O>(
...
-              // eslint-disable-next-line react-native/no-inline-styles
-              style={{ borderBottomWidth: 1 }}
-              onPress={handleOptionPress(item)}
-            >
-              <Typography>{getOptionLabel(item)}</Typography>
-            </PressableStyled>
-          )}
-          data={options}
-        />
+        <FlatList renderItem={renderItem || defaultRenderItem} data={options} />
```
Comment (radoslavzeman) [inline components/inputs/Autocomplete.tsx]: `||` will take all "falsy" values, such as empty string or `0`
In this case, I would use `??` operator, to check only if `renderItem` is `undefined` and `null`.

Generally, `??` is usually what we want to use. And use `||` just in specific cases where you need this specific behaviour.

**Review by radoslavzeman — APPROVED** [review]
Okay, more complex stuff here :D

**Comment by radoslavzeman** [comment]
image

File: components/inputs/Autocomplete.tsx [hunk 2]
```diff
@@ -58,6 +77,8 @@ function AutocompleteInner<O>(
   )
 }
 
-const Autocomplete = forwardRef(AutocompleteInner)
+const Autocomplete = forwardRef(AutocompleteInner) as <O>(
+  p: Props<O> & { ref?: Ref<View> },
+) => ReactElement
```
[the remark this answers is not in the document]
Comment (lukybil) [reply components/inputs/Autocomplete.tsx:110]: it sadly is needed

**Review by radoslavzeman — APPROVED** [review]

**Pull request merged**